tests/components/google_sheets/test_init.py

```diff
@@ -209,11 +212,12 @@ async def test_append_sheet(
     with patch("homeassistant.components.google_sheets.services.Client") as mock_client:
         await hass.services.async_call(
             DOMAIN,
-            "append_sheet",
+            SERVICE_APPEND_SHEET,
             {
-                "config_entry": config_entry.entry_id,
-                "worksheet": "Sheet1",
-                "data": {"foo": "bar"},
+                DATA_CONFIG_ENTRY: config_entry.entry_id,
+                WORKSHEET: "Sheet1",
+                ADD_CREATED_COLUMN: True,
```

Comment:
We should keep a test without the new field, to make sure it defaults to True, but we should validate the resulting sheet if possible.
We should have another call with it set to `False`, and also validate that the column is not added.